ngx-tools/coercion/src/date/date-property.ts

```diff
@@ -12,6 +12,6 @@ export function coerceDateProperty(value: any, fallbackValue = 0) {
  * Whether the provided value is considered a date.
  * @docs-private
  */
-export function _isDateValue(value: any): boolean {
+export function _isDateValue(value: any): value is Date {
```

Comment:
Looking at this makes me wonder if the coerced type will be useful to the consumer since they will be using `coerceDateProperty` rather than `_isDateValue` directly..

I think we can keep this coercion as you have it here (doesn't hurt) but we should also add it to the above overloads (so the consumer gets the benefit):

```typescript
export function coerceDateProperty(value: any): value is Date;
export function coerceDateProperty<D>(value: any, fallback: D): value is Date | D;
export function coerceDateProperty(value: any, fallbackValue = 0) {
  return _isDateValue(value) ? new Date(value) : fallbackValue;
}
```

---

This one got me thinking about testing types like we talked about yesterday.. I wonder if we should create a simple test using an Observable stream where these helpers are used as filters. While it wouldn't really show us via a pass/fail, we _should_ see typescript errors if these helpers are not coercing the types.. What do you think?

Reply:
Thought about adding type guard in overloads, but then thought if it's already guarded in the low level, it wouldn't need at a higher level. But yeah, that doesn't hurt I guess.

The testing idea is interesting. I might dab into it a little. I was just thinking it'd be nice if we could catch that type guard issue in library rather than in core.. and this might help.

Reply:
> if it's already guarded in the low level, it wouldn't need at a higher level

I'm not sure that will work as expected.. I'm pretty sure that TS will be seeing the return type for the parent func if this is used in a pipe etc. (haven't verified though)